core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -1059,9 +1059,10 @@ private List<TemplateParameterInfo> gatherTemplateParamInfo(List<MethodParameter
         for (MethodParameterInfo param : params) {
             List<AnnotationInstance> effectiveParamAnnotations = effectiveParamAnnotations(param);
             if (effectiveParamAnnotations.isEmpty() // if a parameter has no annotations it is considered a template variable
-                    || effectiveParamAnnotations.stream().map(AnnotationInstance::name).anyMatch(MEMORY_ID::equals) // we allow @MemoryId parameters to be
-                                                                                                                    // part of the template
-            ) {
+                    || effectiveParamAnnotations.stream().map(AnnotationInstance::name).anyMatch(dn -> {
+                        // we allow @MemoryId and Camel annotated parameters to be part of the template
+                        return dn.toString().startsWith("org.apache.camel") || MEMORY_ID.equals(dn);
```

Comment:
I'm not sure we want such a hardcoded approach. 
Is there a way to find parameters from jandex?

Reply:
Maybe we can imagine a specific Camel extension that would produce build items we would use here to map parameters?

Reply:
(The reason behind this is that we would need to do the same for the HTTP header and KAfka keys, and it will be very wild quickly.)

Reply:
Yeah, we definitely need something flexible like Clement proposes. 

@aldettinger is there a Camel extension that is used with Quarkus LangChain4j?

Reply:
@cescoffier Ok, it looks there is good incentive for a generic solution right now. On the top my head, let's try to experiment with a `AiServiceMethodParameterAllowedAnnotationBuildItem`. Better name is welcome by the way ;)

@geoand There is already a `camel-quarkus-langchain4j-chat` extension, focusing on chat.  And, I think this pr use case would justify adding a `camel-quarkus-langchain4j` focusing on high level `AiService`. Let's experiment.

Reply:
> And, I think this pr use case would justify adding a camel-quarkus-langchain4j focusing on high level AiService

Right, that's where I was getting at :)

Reply:
Please, find the last update.